### PR TITLE
Adds Capacity Updates and tests to pfs

### LINKED ISSUE #48: PFS receives and processes balance updates from clients
## Description

**As Josh I want to be able to**

get balance proof updates published by the clients processed by the path finding service.

**So that I can**

get a route I can use with high likelihood at the first try.

## Acceptance Criteria

- [x] path finding service listens to the global matrix room for balance proof updates
- [x] path finding service stores and processes all balance proof updates 
- [x] path finding service includes current balances and reveal_timeouts into routing algorithm

## Related story / issue
- [Client publish balance updates to pathfinding service](https://github.com/raiden-network/raiden/issues/3364)
- #47 

**20 SP**

### PR DESCRIPTION
**Pull request by Dominik1999**
closes #48

### CODE REVIEW

File: src/pathfinding_service/exceptions.py [hunk 1]
```diff
@@ -1,6 +1,10 @@
 from typing import Any, Dict, Optional
 
 
+class MessageException(Exception):
+    """Exceptions for incomming messages"""
```
Comment (karlb) [inline src/pathfinding_service/exceptions.py]: Typo: incoming

File: src/pathfinding_service/exceptions.py [hunk 2]
```diff
@@ -1,6 +1,10 @@
 from typing import Any, Dict, Optional
 
 
+class MessageException(Exception):
```
Comment (karlb) [inline src/pathfinding_service/exceptions.py]: This name could be more descriptive. Like `InvalidMessage` or `InvalidBalanceUpdate`.

File: src/pathfinding_service/exceptions.py (hunk 1 above)
Comment (palango) [inline src/pathfinding_service/exceptions.py]: Exception without the s

File: src/pathfinding_service/model/token_network.py [hunk 3]
```diff
@@ -117,6 +117,27 @@ def handle_channel_closed_event(self, channel_identifier: ChannelIdentifier):
                 channel_identifier=channel_identifier,
             )
 
+    def get_channel_view_perspective(
```
Comment (palango) [inline src/pathfinding_service/model/token_network.py]: what's the idea behind *perspective*?

File: src/pathfinding_service/service.py [hunk 4]
```diff
@@ -285,56 +285,87 @@ def create_token_network_for_address(
...
+                    token_network_address=message.canonical_identifier.token_network_address,
+                    channel_identifier=message.canonical_identifier.channel_identifier,
+                    updating_capacity=message.updating_capacity,
+                    other_capacity=message.updating_capacity,
+                )
         else:
             log.info('Ignoring unknown message type')
 
     def on_pfs_update(self, message: UpdatePFS):
+        token_network_address = to_checksum_address(
```
Comment (palango) [inline src/pathfinding_service/service.py]: this should fit in one line

File: src/pathfinding_service/service.py [hunk 5]
```diff
@@ -285,56 +285,87 @@ def create_token_network_for_address(
...
+        if channel_identifier not in token_network.channel_id_to_addresses:
+            raise MessageException(
+                'Received Capacity Update with unknown channel identifier in token network',
             )
-            return
 
         # TODO: check signature of message
 
         # check values max int 256 TODO: get constants
         if not 0 < message.updating_capacity < 2 ^ 256:
```
Comment (palango) [inline src/pathfinding_service/service.py]: can you change this to use the constants?

File: src/pathfinding_service/service.py [hunk 6]
```diff
@@ -285,56 +285,87 @@ def create_token_network_for_address(
...
-            return
+            raise MessageException('Received Capacity Update with impossible capacity')
         if not 0 < message.other_capacity < 2 ^ 256:
-            return
+            raise MessageException('Received Capacity Update with impossible capacity')
+
+        # check if participants fit to channel id
+        if message.updating_participant not in token_network.channel_id_to_addresses[
+            message.canonical_identifier.channel_identifier
```
Comment (palango) [inline src/pathfinding_service/service.py]: This is used below again, can you extract this into a variable?

File: src/pathfinding_service/service.py [hunk 7]
```diff
@@ -285,56 +285,87 @@ def create_token_network_for_address(
...
+            )
+        if message.other_participant not in token_network.channel_id_to_addresses[
+            message.canonical_identifier.channel_identifier
+        ]:
+            raise MessageException(
+                'Other Participant of Capacity Update does not match the internal channel',
+            )
+
+        # check if nonce is higher than current nonce
+
```
Comment (palango) [inline src/pathfinding_service/service.py]: I'd remove the newline here

File: src/pathfinding_service/service.py [hunk 8]
```diff
@@ -285,56 +285,87 @@ def create_token_network_for_address(
...
+        view_to_partner, view_from_partner = token_network.get_channel_view_perspective(
+            channel_identifier=message.canonical_identifier.channel_identifier,
+            updating_participant=message.updating_participant,
+            other_participant=message.other_participant,
+        )
+
+        if (
+            message.updating_nonce <= view_to_partner.update_nonce and
```
Comment (palango) [inline src/pathfinding_service/service.py]: Usually we put the conditions in an extra variable to make the condition easier readable.

File: src/pathfinding_service/model/token_network.py [hunk 9]
```diff
@@ -117,6 +117,27 @@ def handle_channel_closed_event(self, channel_identifier: ChannelIdentifier):
...
+        participant1, participant2 = self.channel_id_to_addresses[channel_identifier]
+        if updating_participant == participant1:
+            assert other_participant == participant2
+            channel_view_to_partner = self.G[participant1][participant2]['view']
+            channel_view_from_partner = self.G[participant2][participant1]['view']
+        else:
+            assert updating_participant == participant2
+            assert other_participant == participant1
+            channel_view_to_partner = self.G[participant2][participant1]['view']
+            channel_view_from_partner = self.G[participant1][participant2]['view']
```
Comment (karlb) [inline src/pathfinding_service/model/token_network.py]: Isn't this just
```python
channel_view_to_partner = self.G[updating_participant][other_participant]['view']
channel_view_from_partner = self.G[other_participant][updating_participant]['view']
```
in both cases?

File: tests/pathfinding/test_capacity_updates.py [hunk 10]
```diff
@@ -0,0 +1,506 @@
+"""
+The tests in this module mock events creation by using a mock blockchain listener.
```
Comment (palango) [inline tests/pathfinding/test_capacity_updates.py]: Is this right?

File: tests/pathfinding/test_capacity_updates.py [hunk 11]
```diff
@@ -0,0 +1,506 @@
...
+
+The Capacity Updates should be tested in various states
```
Comment (palango) [inline tests/pathfinding/test_capacity_updates.py]: That doesn't add so much information.

File: tests/pathfinding/test_capacity_updates.py [hunk 12]
```diff
@@ -0,0 +1,506 @@
...
+
+
+DEFAULT_TOKEN_NETWORK_ADDRESS = Address('0x6e46B62a245D9EE7758B8DdCCDD1B85fF56B9Bc9')
+DEFAULT_TOKEN_NETWORK_ADDRESS_BYTES = TokenNetworkAddress(
+    decode_hex(DEFAULT_TOKEN_NETWORK_ADDRESS),
+)
+DEFAULT_TOKEN_ADDRESS = Address('0x44Ac22fd9672cC559Ab171603D474cEA8a2D7b4D')
+
+"""
+def get_UpdatePFS_message(
```
Comment (palango) [inline tests/pathfinding/test_capacity_updates.py]: Do you still need this, otherwise please remove.

File: tests/pathfinding/test_capacity_updates.py [hunk 13]
```diff
@@ -0,0 +1,506 @@
...
+        return monitor_request
+"""
+
+
+def test_pfs_rejects_capacity_update_with_wrong_chain_id(
+    addresses: List[Address],
+    pathfinding_service_mocked_listeners: PathfindingService,
+):
+
+    message1 = UpdatePFS(
```
Comment (palango) [inline tests/pathfinding/test_capacity_updates.py]: why not call it `message`? Same for all tests below.

File: tests/pathfinding/test_capacity_updates.py [hunk 14]
```diff
@@ -0,0 +1,506 @@
...
+        other_nonce=Nonce(1),
+        updating_capacity=TokenAmount(90),
+        other_capacity=TokenAmount(60),
+        reveal_timeout=2,
+        signature='',
+    )
+
+    with pytest.raises(MessageException) as exinfo:
+        pathfinding_service_mocked_listeners.on_pfs_update(message1)
+    print(str(exinfo.value))
```
Comment (palango) [inline tests/pathfinding/test_capacity_updates.py]: Please remove the print

File: tests/pathfinding/test_capacity_updates.py [hunk 15]
```diff
@@ -0,0 +1,506 @@
...
+        channel_identifier=ChannelIdentifier(0),
+        updating_participant=addresses[0],
+        other_participant=addresses[1],
+    )
+    assert view_to_partner.capacity == 90
+    assert view_to_partner.update_nonce == 1
+    assert view_from_partner.capacity == 110
+    assert view_from_partner.update_nonce == 0
+
+    message8 = UpdatePFS(
```
Comment (palango) [inline tests/pathfinding/test_capacity_updates.py]: might make sense to create a factory function for the `UpdatePFS` messages to reduce duplication.

File: src/pathfinding_service/service.py [hunk 16]
```diff
@@ -285,56 +285,87 @@ def create_token_network_for_address(
...
 
         # check if chain_id matches
-        if message.chain_id != self.chain_id:
-            log.info(
-                'Received balance update with unknown chain id',
-                chain_id=message.chain_id,
-            )
-            return
+        if message.canonical_identifier.chain_identifier != self.chain_id:
+            raise MessageException('Received Capacity Update with unknown chain identifier')
```
Comment (palango) [inline src/pathfinding_service/service.py]: *with*  or *for*? And should we lowercase *capacity update*?

File: src/pathfinding_service/service.py [hunk 17]
```diff
@@ -285,56 +285,87 @@ def create_token_network_for_address(
...
 
         # TODO: check signature of message
 
         # check values max int 256 TODO: get constants
         if not 0 < message.updating_capacity < 2 ^ 256:
-            return
+            raise MessageException('Received Capacity Update with impossible capacity')
         if not 0 < message.other_capacity < 2 ^ 256:
-            return
+            raise MessageException('Received Capacity Update with impossible capacity')
```
Comment (karlb) [inline src/pathfinding_service/service.py]: You could either simplify by doing
```
if (
    (not 0 < message.updating_capacity < 2 ^ 256) or
    (not 0 < message.other_capacity < 2 ^ 256)
):
    raise MessageException('Received Capacity Update with impossible capacity')
```
or make the errors more informative by adding "impossible `updating_capacity`" and "impossible `other_capacity`" if you keep the two `raise` statements.

File: src/pathfinding_service/exceptions.py (hunk 1 above)
Comment (Dominik1999) [reply src/pathfinding_service/exceptions.py]: done

File: src/pathfinding_service/service.py [hunk 18]
```diff
@@ -285,56 +285,87 @@ def create_token_network_for_address(
...
+        # check if participants fit to channel id
+        if message.updating_participant not in token_network.channel_id_to_addresses[
+            message.canonical_identifier.channel_identifier
+        ]:
+            raise MessageException(
+                'Sender of Capacity Update does not match the internal channel',
+            )
+        if message.other_participant not in token_network.channel_id_to_addresses[
+            message.canonical_identifier.channel_identifier
+        ]:
```
Comment (karlb) [inline src/pathfinding_service/service.py]: Not necessary, but IMO a bit nicer would be to extract
```python
participants = token_network.channel_id_to_addresses[message.canonical_identifier.channel_identifier]
if message.updating_participant not in participants:
    raise ...
if if message.other_participant not in participants:
    raise ...
```

File: src/pathfinding_service/model/token_network.py (hunk 3 above)
Comment (Dominik1999) [reply src/pathfinding_service/model/token_network.py]: I thought the channel view with respect to the perspective of the updating participant, makes sense?

File: src/pathfinding_service/service.py (hunk 4 above)
Comment (Dominik1999) [reply src/pathfinding_service/service.py]: to me it looks like it would be 104 chars in the line then

Comment (palango) [reply src/pathfinding_service/service.py]: then nevermind

File: src/pathfinding_service/service.py (hunk 5 above)
Comment (Dominik1999) [reply src/pathfinding_service/service.py]: done, we now have a `MAX_TOKEN_AMOUNT` in config

File: src/pathfinding_service/service.py (hunk 6 above)
Comment (Dominik1999) [reply src/pathfinding_service/service.py]: done

File: src/pathfinding_service/service.py (hunk 8 above)
Comment (Dominik1999) [reply src/pathfinding_service/service.py]: I do not quite understand what you mean. Maybe like this?
```
       a = view_to_partner.update_nonce
       b = view_from_partner.update_nonce
      if (
            message.updating_nonce <= a and
            message.other_nonce <= b
        ):
```

File: tests/pathfinding/test_capacity_updates.py (hunk 10 above)
Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: good point. changed it

File: tests/pathfinding/test_capacity_updates.py (hunk 11 above)
Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: reformulated into 
`The Capacity Updates show different correct and incorrect values to test all edge cases`

File: src/pathfinding_service/service.py (hunk 5 above)
Comment (karlb) [reply src/pathfinding_service/service.py]: There's also `raiden.constants.UINT256_MAX`. If you want to use that, use `<=`, as it's `2 ** 256 - 1`.

Please note that `2 ^ 256` is not what you want in this case:
```python
>>> 2 ^ 256
258
```
The `^` operator is a bitwise or.

File: tests/pathfinding/test_capacity_updates.py (hunk 12 above)
Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: done

File: tests/pathfinding/test_capacity_updates.py (hunk 14 above)
Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: good point. refactored

File: src/pathfinding_service/service.py (hunk 8 above)
Comment (palango) [reply src/pathfinding_service/service.py]: More like:
```
      valid nonce = (
         message.updating_nonce <= view_to_partner.update_nonce and
         message.other_nonce <= view_from_partner.update_nonce
      if valid_nonce:

File: tests/pathfinding/test_capacity_updates.py (hunk 13 above)
Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: done

File: src/pathfinding_service/service.py (hunk 16 above)
Comment (Dominik1999) [reply src/pathfinding_service/service.py]: the message shows an unknown chain ID. So I would say received something "with"

Comment (Dominik1999) [reply src/pathfinding_service/service.py]: imho "Capacity Update" should be uppercase since we use it as a term?!

File: src/pathfinding_service/service.py (hunk 5 above)
Comment (Dominik1999) [reply src/pathfinding_service/service.py]: good catch, we now use UINT256_MAX ... also in the tests

File: src/pathfinding_service/exceptions.py (hunk 1 above)
Comment (Dominik1999) [reply src/pathfinding_service/exceptions.py]: indeed

File: src/pathfinding_service/exceptions.py (hunk 2 above)
Comment (Dominik1999) [reply src/pathfinding_service/exceptions.py]: I would go for `InvalidCapacityUpdate` ?!

File: src/pathfinding_service/service.py (hunk 17 above)
Comment (Dominik1999) [reply src/pathfinding_service/service.py]: that is a good one, I kept the two raises but made them more informative

File: src/pathfinding_service/model/token_network.py (hunk 9 above)
Comment (Dominik1999) [reply src/pathfinding_service/model/token_network.py]: to be honest, yes...

File: src/pathfinding_service/service.py (hunk 18 above)
Comment (Dominik1999) [reply src/pathfinding_service/service.py]: good point... changed it. I think we need this check, especially when we don't check this later when we get the channel views

File: src/pathfinding_service/service.py (hunk 8 above)
Comment (Dominik1999) [reply src/pathfinding_service/service.py]: understood, changed it

File: src/pathfinding_service/model/token_network.py (hunk 3 above)
Comment (palango) [reply src/pathfinding_service/model/token_network.py]: What about ` get_channel_views_for_partner`?

Comment (Dominik1999) [reply src/pathfinding_service/model/token_network.py]: good suggestion changed it

File: tests/pathfinding/test_capacity_updates.py (hunk 15 above)
Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: done

Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: done

File: src/pathfinding_service/service.py [hunk 19]
```diff
@@ -285,56 +285,88 @@ def create_token_network_for_address(
...
-            return
 
         # TODO: check signature of message
 
         # check values max int 256 TODO: get constants
-        if not 0 < message.updating_capacity < 2 ^ 256:
-            return
-        if not 0 < message.other_capacity < 2 ^ 256:
-            return
+        if not 0 < message.updating_capacity < UINT256_MAX:
```
Comment (LefterisJP) [inline src/pathfinding_service/service.py]: you can now remove the comment from above as you are using the constants.

Also the comparison here and below is confusing. Do not use not with the comparison operators. Simply reverse them. Additionally isn't UINT256_MAX a valid value? So shouldn't it be `<=` there?

File: src/pathfinding_service/service.py [hunk 20]
```diff
@@ -285,56 +285,88 @@ def create_token_network_for_address(
...
+            raise InvalidCapacityUpdate(
+                'Received Capacity Update with impossible updating_capacity',
+            )
+        if not 0 < message.other_capacity < UINT256_MAX:
```
Comment (LefterisJP) [inline src/pathfinding_service/service.py]: same as above

File: src/pathfinding_service/service.py [hunk 21]
```diff
@@ -285,56 +285,88 @@ def create_token_network_for_address(
...
+            raise InvalidCapacityUpdate(
+                'Received Capacity Update with impossible other_capacity',
```
Comment (LefterisJP) [inline src/pathfinding_service/service.py]: nitpick: Here and other parts of the file, why capitalize the words capacity and update?

File: tests/pathfinding/test_capacity_updates.py [hunk 22]
```diff
@@ -0,0 +1,409 @@
...
+
+
+DEFAULT_TOKEN_NETWORK_ADDRESS = Address('0x6e46B62a245D9EE7758B8DdCCDD1B85fF56B9Bc9')
+DEFAULT_TOKEN_NETWORK_ADDRESS_BYTES = TokenNetworkAddress(
+    decode_hex(DEFAULT_TOKEN_NETWORK_ADDRESS),
+)
+DEFAULT_TOKEN_ADDRESS = Address('0x44Ac22fd9672cC559Ab171603D474cEA8a2D7b4D')
+
+
+def get_UpdatePFS_message(
```
Comment (LefterisJP) [inline tests/pathfinding/test_capacity_updates.py]: The naming here looks wrong. I get it's to get the class but if you go for snake case, please stick to snake case. Mixing Camelcase and snake_case in the same object name looks wrong.

File: tests/pathfinding/test_capacity_updates.py [hunk 23]
```diff
@@ -0,0 +1,409 @@
...
+
+    message = get_UpdatePFS_message(
+        chain_identifier=ChainID(121212),
+        updating_participant=addresses[0],
+        other_participant=addresses[1],
+    )
+
+    with pytest.raises(InvalidCapacityUpdate) as exinfo:
+        pathfinding_service_mocked_listeners.on_pfs_update(message)
+    print(str(exinfo.value))
```
Comment (LefterisJP) [inline tests/pathfinding/test_capacity_updates.py]: Print is still here, at least that's what the github diff shows me

File: tests/pathfinding/test_capacity_updates.py [hunk 24]
```diff
@@ -0,0 +1,409 @@
...
+
+    message = get_UpdatePFS_message(
+        channel_identifier=ChannelID(35),
+        updating_participant=addresses[0],
+        other_participant=addresses[1],
+    )
+
+    with pytest.raises(InvalidCapacityUpdate) as exinfo:
+        pathfinding_service_mocked_listeners.on_pfs_update(message)
+    print(str(exinfo.value))
```
Comment (LefterisJP) [inline tests/pathfinding/test_capacity_updates.py]: please remove print statement

File: tests/pathfinding/test_capacity_updates.py [hunk 25]
```diff
@@ -0,0 +1,409 @@
...
+
+    message = get_UpdatePFS_message(
+        updating_participant=addresses[0],
+        other_participant=addresses[1],
+        updating_capacity=TokenAmount(UINT256_MAX + 1),
+    )
+
+    with pytest.raises(InvalidCapacityUpdate) as exinfo:
+        pathfinding_service_mocked_listeners.on_pfs_update(message)
+    print(str(exinfo.value))
```
Comment (LefterisJP) [inline tests/pathfinding/test_capacity_updates.py]: remove print statement

File: tests/pathfinding/test_capacity_updates.py [hunk 26]
```diff
@@ -0,0 +1,409 @@
...
+
+    message = get_UpdatePFS_message(
+        updating_participant=addresses[0],
+        other_participant=addresses[1],
+        other_capacity=TokenAmount(UINT256_MAX + 1),
+    )
+
+    with pytest.raises(InvalidCapacityUpdate) as exinfo:
+        pathfinding_service_mocked_listeners.on_pfs_update(message)
+    print(str(exinfo.value))
```
Comment (LefterisJP) [inline tests/pathfinding/test_capacity_updates.py]: remove print statement

File: tests/pathfinding/test_capacity_updates.py [hunk 27]
```diff
@@ -0,0 +1,409 @@
...
+    )
+
+    message = get_UpdatePFS_message(
+        updating_participant=addresses[2],
+        other_participant=addresses[1],
+    )
+
+    with pytest.raises(InvalidCapacityUpdate) as exinfo:
+        pathfinding_service_mocked_listeners.on_pfs_update(message)
+    print(str(exinfo.value))
```
Comment (LefterisJP) [inline tests/pathfinding/test_capacity_updates.py]: remove print statement

File: tests/pathfinding/test_capacity_updates.py [hunk 28]
```diff
@@ -0,0 +1,409 @@
...
+    )
+
+    message = get_UpdatePFS_message(
+        updating_participant=addresses[0],
+        other_participant=addresses[2],
+    )
+
+    with pytest.raises(InvalidCapacityUpdate) as exinfo:
+        pathfinding_service_mocked_listeners.on_pfs_update(message)
+    print(str(exinfo.value))
```
Comment (LefterisJP) [inline tests/pathfinding/test_capacity_updates.py]: remove print statement. I like repeating myself :smile:

File: tests/pathfinding/test_capacity_updates.py [hunk 29]
```diff
@@ -0,0 +1,409 @@
...
+    )
+    assert view_to_partner.capacity == 90
+    assert view_to_partner.update_nonce == 1
+    assert view_from_partner.capacity == 110
+    assert view_from_partner.update_nonce == 0
+
+    # Send the same Capacity Update again - leads to an exception
+    with pytest.raises(InvalidCapacityUpdate) as exinfo:
+        pathfinding_service_mocked_listeners.on_pfs_update(message)
+    print(str(exinfo.value))
```
Comment (LefterisJP) [inline tests/pathfinding/test_capacity_updates.py]: remove print statement. I like repeating myself a lot  :smile:

File: tests/pathfinding/test_capacity_updates.py [hunk 30]
```diff
@@ -0,0 +1,409 @@
...
+    assert 'Other Participant of Capacity Update does not match' in str(exinfo.value)
+
+
+def test_pfs_rejects_capacity_update_with_wrong_nonces(
+        addresses: List[Address],
+        pathfinding_service_mocked_listeners: PathfindingService,
+):
+    pathfinding_service_mocked_listeners.chain_id = 1
```
Comment (LefterisJP) [inline tests/pathfinding/test_capacity_updates.py]: So looking at all these tests, there is a **lot of** duplicated code. Can't you abstract away the setup of the tests into a common function and then use it in all of the tests?

File: src/pathfinding_service/service.py (hunk 19 above)
Comment (Dominik1999) [reply src/pathfinding_service/service.py]: nice catch, changed it

File: src/pathfinding_service/service.py (hunk 21 above)
Comment (Dominik1999) [reply src/pathfinding_service/service.py]: We wanted to use "Capacity Update" as a term ... The message should be called Capacity Update. Does that make sense to you?

File: tests/pathfinding/test_capacity_updates.py (hunk 22 above)
Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: I am a bit a newbee, but wtf is a camel and a snake case?

File: tests/pathfinding/test_capacity_updates.py (hunk 26 above)
Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: thanks

File: tests/pathfinding/test_capacity_updates.py (hunk 27 above)
Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: thanks

File: tests/pathfinding/test_capacity_updates.py (hunk 29 above)
Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: thanks

File: tests/pathfinding/test_capacity_updates.py (hunk 28 above)
Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: thanks

File: tests/pathfinding/test_capacity_updates.py (hunk 30 above)
Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: This is a good point. I do not have much time today refactoring this. I will open an issue and get this done next week. Would that be sufficient?

File: src/pathfinding_service/service.py (hunk 21 above)
Comment (LefterisJP) [reply src/pathfinding_service/service.py]: Sure I get it. It just seems odd a bit. But it's fine. Just felt the need to ask if it's a typo/problem.

File: tests/pathfinding/test_capacity_updates.py (hunk 22 above)
Comment (LefterisJP) [reply tests/pathfinding/test_capacity_updates.py]: @Dominik1999 

**Snake case**: `def a_function_that_adds_two_numbers()`
**Camel case**: `def AFunctionThatAddsTwoNumbers()`

They are two different naming style methods. We don't mix those.

https://blog.lmorchard.com/2013/01/23/naming-conventions/

File: tests/pathfinding/test_capacity_updates.py (hunk 30 above)
Comment (LefterisJP) [reply tests/pathfinding/test_capacity_updates.py]: Sure, as long as we have a task for it this is fine.

File: src/pathfinding_service/service.py (hunk 21 above)
Comment (Dominik1999) [reply src/pathfinding_service/service.py]: Yes good that you pointed that out!

File: tests/pathfinding/test_capacity_updates.py (hunk 22 above)
Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: a now I get it, changed the function name

File: tests/pathfinding/test_capacity_updates.py (hunk 30 above)
Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: ok

Comment (Dominik1999) [reply tests/pathfinding/test_capacity_updates.py]: #200

**Comment by LefterisJP** [comment]
@Dominik1999 Looks good to me. I don't know what kind of rebasing/merging strategies you use in this repo but it would be nice to avoid merging commits and use rebase instead.
